### ISSUE TITLE
Some cipher suites are missing in the result

### ISSUE THREAD

**Issue by jensguballa** [nabla-c0d3/sslyze#382]
Performing a scan against a certain domain I get the impression that sslyze does not report all  cipher suites, e.g. for TLS1.2.

For example, using Firefox (disabling TLS1.3 first) on that domain the browser tells me that the connection is protected using TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 with TLS1.2). This cipher suite is not reported by sslyze.

It seems to me, that at least the following cipher suites are not reported by sslyze:
- TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
- TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
- TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA
- TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA

Other scan tools do list those cipher suites (tested with testssl.sh, ssllabs).

Here is my setup:
sslyze version: 2.1.3
openssl version: 1.1.1c

**Comment by nabla-c0d3** [comment]
Can you post (or send me) a domain where you have seen this behavior? Thanks!

**Comment by jensguballa** [comment]
I observed this behavior for google.com. Other scanners like sslllabs report those cipher suites which I think are missing in sslyze.

**Comment by dhaug-op** [comment]
I tried to dig into this. The problem seems to be related to https://github.com/nabla-c0d3/sslyze/blob/master/sslyze/utils/tls12_workaround.py

I could find out, that the listed ciphers are listed in ``legacy_ciphers`` and thus return ``requires_legacy_openssl`` with ``True``. But the legacy openssl seems to not be able to use the cipher. I also saw this behavior with a plain installation of OpenSSL 1.0.2e:

```bash
vagrant@kali:~$ /usr/local/ssl/bin/openssl version
OpenSSL 1.0.2e 3 Dec 2015
vagrant@kali:~$ /usr/local/ssl/bin/openssl ciphers -v | grep "ECDHE-ECDSA-AES256-SHA"
ECDHE-ECDSA-AES256-SHA384 TLSv1.2 Kx=ECDH     Au=ECDSA Enc=AES(256)  Mac=SHA384
ECDHE-ECDSA-AES256-SHA  SSLv3 Kx=ECDH     Au=ECDSA Enc=AES(256)  Mac=SHA1
vagrant@kali:~$ /usr/local/ssl/bin/openssl s_client -cipher 'ECDHE-ECDSA-AES256-SHA' -connect google.com:443
CONNECTED(00000003)
139731961992896:error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure:s23_clnt.c:769:
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 7 bytes and written 139 bytes
---
New, (NONE), Cipher is (NONE)
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : 0000
    Session-ID:
    Session-ID-ctx:
    Master-Key:
    Key-Arg   : None
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    Start Time: 1569853579
    Timeout   : 300 (sec)
    Verify return code: 0 (ok)
---
vagrant@kali:~$
```

Unfortunately do not know, why this is the case. But for ``google.com`` I got all results correctly, when I reverted https://github.com/nabla-c0d3/sslyze/commit/dd5c5af7c251a904de5213639637bb6ddd3959b6 to use the explicit list of ciphers in the ``requires_legacy_openssl``-function.

If anybody can provide more insights into this I would be happy to help resolve this issue, as I like SSLyze and this a huge problem.

Thank you all.

**Comment by nabla-c0d3** [comment]
Fixed in v3.0.0.
